src/test/java/com/squareup/javapoet/MethodSpecTest.java

```diff
@@ -177,6 +182,12 @@ private ExecutableElement findFirst(Collection<ExecutableElement> elements, Stri
         + "@java.lang.Override\n"
         + "public int compareTo(" + ExtendsOthers.class.getCanonicalName() + " arg0) {\n"
         + "}\n");
+    exec = findFirst(methods, "fail");
+    method = MethodSpec.overriding(exec, classType, types).build();
+    assertThat(method.toString()).isEqualTo(""
+        + "@java.lang.Override\n"
+        + "public void fail() throws java.lang.IllegalStateException {\n"
```

Comment:
instead of `throws  R`.
i'll try to remember to check KotlinPoet this weekend, too.

Reply:
there's no way to represent
```java
interface Throws<R extends RuntimeException> {
  void fail() throws R;
}
```
in Kotlin, as far as I can tell!